### ISSUE TITLE
Controller is not a "specialized Component" (documentation issue)

### ISSUE THREAD

**Issue by locks** [emberjs/guides#1725]
> Controller is an Object, while Component is a View. Controller has no idea it has some DOM elements attached to it and it won't know its this.$() for example. It seems it's also unable to process events passed by actions. It's not "just a big Component" and shouldn't be described as such.
> If I'm wrong in some way please explain. I'm fairly new to Ember.

@Senthe

**Comment by locks** [comment]
@Senthe by specialized what is meant is that Controller has a narrower scope than a Component, meaning it doesn't have access to the DOM like you mention. Can you suggest a clearer way to state the previous so we can update the respective Guide?

> It seems it's also unable to process events passed by actions.

I don't entirely understand this point.

**Comment by Senthe** [comment]
In Component I can define `myAction(event)` and use event object from browser, click event for example. In Controller `myAction(event)` will have undefined `event` because Controller doesn't get anything from browser/DOM apparently.

When I see "specialized type" of something I translate it to "extends". "Webdeveloper is a specialized type of Developer" = Webdeveloper IS a Developer who has some additional abilities, is a specialist.

But Controller is not more than Component. It's less than Component, can't do what Component can. So it's actually something completely else, not Components child, but a sibling.

**Comment by locks** [comment]
Can you reproduce your problem in an [Ember Twiddle](http://ember-twiddle.com)? If you're using actions as DOM event handlers it shouldn't matter if the handler is in a Component or Controller, it should still receive the event.

**Comment by Senthe** [comment]
Okay, I will try then. Maybe the issue is my Emblem syntax, but when I use exactly the same syntax, it works for Component but not for Controller.

**Comment by Senthe** [comment]
I tried reproducing it: https://ember-twiddle.com/6836aac7cc251aa82b7914ebaacc964d?openFiles=templates.application.hbs%2Ctemplates.components.my-component.hbs

But this time I actually get two "undefined". I'm not sure what I did wrong. I use Emblem in my project so I don't know if syntax I use there is exactly correct.

**Comment by acorncom** [comment]
I know this may be a tangent, but we would generally recommend that people don't use Emblem (especially if you are just starting an Ember project).  It makes it harder to work with other Ember devs (as you've just seen) and might lead to syntax troubles in the long run ...

Back to the question at hand, @Senthe is there some place in the guides that was what led to your initial confusion here?  Or was it something else you read?

**Comment by Senthe** [comment]
Okay, Controllers guide states:

>Controllers behave like a specialized type of Component that is rendered by the router when entering a Route.

So I would think "oh okay, so Controller is basically a Component that gets created for every route - neat!".

But this is not true.

* Controller is a bare Object. Component is a View.
* Controller doesn't have Lifecycle Hooks. Component has them.
* Controller is abstract structure/singleton. Component is always hooked to DOM and every component is a standalone instance of some structure.
* You test them differently as well. You will want probably to render Component to test it, while Controller is abstract and can/should be tested without rendering anything.
* etc. etc. add any other difference between View and Object.

Basically, Controller is anything BUT subtype of Component.

**Comment by theroncross** [comment]
@Senthe I can definitely see your point here. More specifically, a Controller plays a similar role to the .js file of a controller, with the addition of being passed a model object by the route. Would that be a better explanation? I'm happy to make the change if you thing it helps.

**Comment by Senthe** [comment]
@theroncross hello again after a couple months : ). I got much better grasp on Ember and am happy to discuss the issue. 

Right now this is the introduction to Controllers:

> Controllers behave like a specialized type of Component that is rendered by the router when entering a Route.
> 
> The controller receives a single property from the Route – model – which is the return value of the Route's model() method.
> 
> To define a Controller, run: `<code>`
> 
> The value of my-controller-name must match the name of the Route that renders it. So a Route named blog-post would have a matching Controller named blog-post.
> 
> You only need to generate a Controller if you want to customize its properties or provide any actions. If you have no customizations, Ember will provide a Controller instance for you at run time.

I would rephrase a bit to something like: 

> **Controllers are created when entering a Route and contain properties and actions specific to their Routes.** 
> 
> You only need to generate a Controller if you want to customize its properties or provide any actions. If you have no customizations, Ember will provide a Controller instance for you at run time.
>
> To define a Controller, run: `<code>`
> 
> The value of my-controller-name must match the name of the Route that renders it. So a Route named blog-post would have a matching Controller named blog-post.
> 
> The controller receives a single property from the Route – model – which is the return value of the Route's model() method.

How does that look to you?

Shouldn't it also be documented in this section that Controllers are actually singletons reused on route enter, that can only be re-"rendered" in `setupController` hook?
